From: olh@suse.de
subject: fix compile errors.

In file included from sound/sound_core.c:44:
include/linux/sound.h:33: warning: 'struct device' declared inside parameter list
include/linux/sound.h:33: warning: its scope is only this definition or declaration, which is probably not what you want
sound/sound_core.c:242: error: conflicting types for 'register_sound_special_device'
include/linux/sound.h:33: error: previous declaration of 'register_sound_special_device' was here

 include/linux/sound.h |    1 +
 1 files changed, 1 insertion(+)

Index: linux-2.6.12/include/linux/sound.h
===================================================================
--- linux-2.6.12.orig/include/linux/sound.h
+++ linux-2.6.12/include/linux/sound.h
@@ -29,6 +29,7 @@
  *	Sound core interface functions
  */
  
+struct device;
 extern int register_sound_special(struct file_operations *fops, int unit);
 extern int register_sound_special_device(struct file_operations *fops, int unit, struct device *dev);
 extern int register_sound_mixer(struct file_operations *fops, int dev);
